specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1740 @@
+          "type": "number",
+          "format": "decimal",
+          "readOnly": true
+        },
+        "maxUtilizationPercentage": {
+          "description": "This is the maximum hourly utilization for each date range that corresponds to given grain (Daily, Monthly). Suppose the API call is for usageDate > 2023-03-01 and usageDate < 2022-10-31 at a daily granularity. There will be one record per benefit id for each day. For a single day, the maxUtilizationPercentage value will be equal to the largest in the set of values where the set contains 24 utilization percentage entries one for each hour in a specific day. If on the day 2022-10-18, the largest utilization percentage was 90% at hour 5, then the value for the maxUtilizationPercentage in the response will be 90%.",
+          "type": "number",
+          "format": "decimal",
+          "readOnly": true
+        }
+      }
+    },
+    "BenefitUtilizationSummariesOperationStatus": {
+      "type": "object",
+      "description": "Status of a benefit utilization summaries report. Provides Async Benefit Utilization Summaries Request input, status, and report sas url.",
+      "properties": {
+        "input": {
+          "$ref": "#/definitions/BenefitUtilizationSummariesRequest",
+          "description": "Input given to create the benefit utilization summaries report."
+        },
+        "status": {
+          "description": "The status of the creation of the benefit utilization summaries report.",
+          "$ref": "#/definitions/AsyncOperationStatusEnum"
+        },
+        "properties": {
+          "description": "Contains sas url to the async benefit utilization summaries report and a date that the url is valid until. These values will be empty if the report is in a Running or Failed state",
+          "$ref": "#/definitions/AsyncOperationStatusProperties"
+        }
+      }
+    },
+    "AsyncOperationStatusEnum": {
+      "description": "Enum representing the status of an async operation.",
+      "type": "string",
+      "enum": [
+        "Running",
+        "Complete",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "OperationStatusType",
+        "modelAsString": true
+      }
+    },
+    "AsyncOperationStatusProperties": {
+      "description": "Object representing the report url and valid until date of the async report generated.",
+      "type": "object",
+      "properties": {
+        "reportUrl": {
+          "description": "Sas url to the async benefit utilization summaries report. Will be empty if the report is in Running or Failed state.",
+          "$ref": "#/definitions/BenefitUtilizationSummaryReportSchema"
+        },
```

Comment:
the name and description of the property says Url, but it links to a string enum and this confuses me.

(Also, SAS URLs contain a security token, and I imagine in this case it grants some read access, so I think these should be marked as x-ms-secret if they are actually SAS uris)

Reply:
The enum description clarifies these values are the column names of the report. This is consistent with how we have previously defined our Reservation Details report schema which can be found in the costmanagement.json file. I will add x-ms-secret

Reply:
OK. I have a feeling the SDK autorest stuff is going to be really confused by doing it this way though.